src/events/mongodb.js

```diff
@@ -24,21 +24,49 @@ function stringifyData(obj) {
     return data.substring(0, MAX_DATA_LENGTH);
 }
 
+/**
+ * Extracts connection details from a conenction
+ * @param {Object} connId The connectionId object of the instrumented event
+ * @return {{port: *, host: *}} extracted connection details
+ */
+function getConnectionDetails(connId) {
+    let host;
+    let port;
+    utils.debugLog('Epsagon - MongoDB - inspecting connectionId', connId);
+    if (connId) {
+        if (typeof connId === 'string') {
+            const parts = connId.split(':');
+            if (parts.length && parts[0][0] === '/') {
+                host = 'localhost';
+                [port] = parts;
+            } else {
+                [host, port] = parts;
+            }
+        } else if (connId.domainSocket) {
+            [host, port] = ['localhost', connId.host]; // host for domainSocket is the identifier
+        } else {
+            // eslint-disable-next-line prefer-destructuring
+            host = connId.host;
+            // eslint-disable-next-line prefer-destructuring
+            port = connId.port;
+        }
+    }
+    return { host, port };
+}
+
 /**
  * Hook for mongodb requests starting
  * @param {Object} event The request event
  */
 function onStartHook(event) {
     try {
         const startTime = Date.now();
+        utils.debugLog('Epsagon - MongoDB - handling event', event);
 
-        let { host, port } = event.connectionId;
-        if (!host) {
-            [host, port] = event.connectionId.split(':');
-        }
+        const { host, port } = getConnectionDetails(event ? event.connectionId : null);
```

Comment:
If there's no event is there any point to continue at all? Maybe check if event.connectionId?

Reply:
not really possible, just a safeguard